src/qt/proposallist.cpp

```diff
@@ -268,45 +268,69 @@ ProposalList::ProposalList(QWidget *parent) :
-    proposalList->setColumnWidth(ProposalTableModel::YesVotes, YES_VOTES_COLUMN_WIDTH);
-    proposalList->setColumnWidth(ProposalTableModel::NoVotes, NO_VOTES_COLUMN_WIDTH);
-    proposalList->setColumnWidth(ProposalTableModel::AbstainVotes, ABSTAIN_COLUMN_WIDTH);
-    proposalList->setColumnWidth(ProposalTableModel::VotesNeeded, VOTES_NEEDED_COLUMN_WIDTH);
-
-    columnResizingFixer = new GUIUtil::TableViewLastColumnResizingFixer(proposalList, VOTES_NEEDED_COLUMN_WIDTH, MINIMUM_COLUMN_WIDTH);
-
     nLastUpdate = GetTime();
 
     timer = new QTimer(this);
     connect(timer, SIGNAL(timeout()), this, SLOT(refreshProposals()));
     timer->start(1000);
+}
+
+void ProposalList::setModel(WalletModel* model)
+{
 
-    setLayout(vlayout);
+    QSettings settings;
+    proposalTableModel = new ProposalTableModel(this);
+    this->model = model;
+
+    if (model) {
+        proposalProxyModel = new ProposalFilterProxy(this);
+        proposalProxyModel->setDynamicSortFilter(true);
+        proposalProxyModel->setFilterCaseSensitivity(Qt::CaseInsensitive);
+        proposalProxyModel->setSortCaseSensitivity(Qt::CaseInsensitive);
+        proposalProxyModel->setSortRole(Qt::EditRole);
+        proposalProxyModel->setSourceModel(proposalTableModel);
+
+        proposalList->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+        proposalList->setModel(proposalProxyModel);
+        proposalList->setAlternatingRowColors(true);
+        proposalList->setSelectionBehavior(QAbstractItemView::SelectRows);
+        proposalList->setSortingEnabled(true);
+        proposalList->sortByColumn(ProposalTableModel::StartDate, Qt::DescendingOrder);
+        proposalList->verticalHeader()->hide();
+
+        proposalList->setColumnWidth(ProposalTableModel::Proposal, PROPOSAL_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::Amount, AMOUNT_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::StartDate, START_DATE_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::EndDate, END_DATE_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::TotalPaymentCount, TOTAL_PAYMENT_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::RemainingPaymentCount, REMAINING_PAYMENT_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::YesVotes, YES_VOTES_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::NoVotes, NO_VOTES_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::AbstainVotes, ABSTAIN_COLUMN_WIDTH);
+        proposalList->setColumnWidth(ProposalTableModel::VotesNeeded, VOTES_NEEDED_COLUMN_WIDTH);
+
+        columnResizingFixer = new GUIUtil::TableViewLastColumnResizingFixer(proposalList, VOTES_NEEDED_COLUMN_WIDTH, MINIMUM_COLUMN_WIDTH);
+    }
 }
 
 void ProposalList::createProposal()
 {
+
+    WalletModel::EncryptionStatus encStatus = model->getEncryptionStatus();
```

Comment:
Though it works, it's better to add nullptr check for `model` before using it.

Reply:
We could probably do that, this is just how it was done in the Masternode page